File: web/lib/regions.ts
// vlr.gg's rankings regions — the single source of truth for the UI.
// Mirrors src/vlr/regions.py so the app's regional views match vlr.gg exactly.

export interface VlrRegion {
  slug: string;
  label: string;
  short: string;
}

export const VLR_REGIONS: VlrRegion[] = [
  { slug: 'north-america', label: 'North America', short: 'NA' },
  { slug: 'europe', label: 'Europe', short: 'EU' },
  { slug: 'brazil', label: 'Brazil', short: 'BR' },
  { slug: 'asia-pacific', label: 'Asia-Pacific', short: 'AP' },
  { slug: 'korea', label: 'Korea', short: 'KR' },
  { slug: 'china', label: 'China', short: 'CN' },
  { slug: 'japan', label: 'Japan', short: 'JP' },
  { slug: 'la-s', label: 'LA-S', short: 'LAS' },
  { slug: 'la-n', label: 'LA-N', short: 'LAN' },
  { slug: 'oceania', label: 'Oceania', short: 'OCE' },
  { slug: 'mena', label: 'MENA', short: 'MN' },
  { slug: 'gc', label: 'Game Changers', short: 'GC' },
  { slug: 'collegiate', label: 'Collegiate', short: 'CG' },
];

// Human label for a region slug (falls back to the slug itself).
export function regionLabel(slug: string | null | undefined): string {
  if (!slug) return '—';
  return VLR_REGIONS.find((r) => r.slug === slug)?.label ?? slug;
}

File: src/vlr/regions.py
"""Canonical vlr.gg rankings regions — the single source of truth.

vlr.gg/rankings exposes one page per region (``/rankings/<slug>``). We scrape
each one and store its slug directly on the team, so every regional view in the
app mirrors vlr.gg exactly instead of collapsing everything into a handful of
franchise buckets.

Order matches vlr's own region selector.
"""
from __future__ import annotations

# (slug, display label, short code) in vlr's display order.
VLR_REGIONS: list[tuple[str, str, str]] = [
    ("north-america", "North America", "NA"),
    ("europe", "Europe", "EU"),
    ("brazil", "Brazil", "BR"),
    ("asia-pacific", "Asia-Pacific", "AP"),
    ("korea", "Korea", "KR"),
    ("china", "China", "CN"),
    ("japan", "Japan", "JP"),
    ("la-s", "LA-S", "LAS"),
    ("la-n", "LA-N", "LAN"),
    ("oceania", "Oceania", "OCE"),
    ("mena", "MENA", "MN"),
    ("gc", "Game Changers", "GC"),
    ("collegiate", "Collegiate", "CG"),
]

# Slugs in scrape / display order.
VLR_REGION_SLUGS: list[str] = [slug for slug, _label, _short in VLR_REGIONS]
VLR_REGION_SET: frozenset[str] = frozenset(VLR_REGION_SLUGS)
